Util/channel/src/main/java/io/deephaven/util/channel/DataInstructionsProviderPlugin.java

```diff
@@ -0,0 +1,20 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.util.channel;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * A plugin interface for providing {@link DataInstructionsProviderPlugin} implementations for different property
+ * collections and URI values. Check out {@link DataInstructionsProviderLoader} for more details.
+ */
+public interface DataInstructionsProviderPlugin {
```

Comment:
I think scoping these in Util ATM is too broad. I think we should move it into something like `io.deephaven.iceberg.internal`.